src/shared/bootspec.c

```diff
@@ -853,6 +863,123 @@ static int find_uki_sections(
         return 0;
 }
 
+static int find_addon_sections(
+                int fd,
+                const char *path,
+                char **ret_cmdline) {
+
+        _cleanup_free_ IMAGE_SECTION_HEADER *sections = NULL;
+        _cleanup_free_ PeHeader *pe_header = NULL;
+        int r;
+
+        r = find_sections(fd, path, &sections, &pe_header);
+        if (r < 0)
+                return r;
+
+        return find_cmdline_section(fd, path, sections, pe_header, ret_cmdline);
+}
+
+static int insert_boot_entry_addon_array(
```

Comment:
We have some helpers for this. Also, it is strange that `addons` is the last parameter. We are modifying `addons` here, so it should be the first parameter. There is some discussion of this in CODING_STYLE.

Thus:

```C
static int insert_boot_entry_addon(
      BootEntryAddons *addons,
      char *location,
      char *cmdline) {
if (!GREEDY_REALLOC(addons->items, addons->count + 1))
     return log_oom();
addons->items[addons->count] = (BootEntryAddon) {
  .location = location;
  .cmdline = cmdline;
};
addons->count++;
return 0;
```